biz.aQute.resolve/test/biz/aQute/resolve/obr/plugin/ObrExporterTest.java

```diff
@@ -0,0 +1,150 @@
+package biz.aQute.resolve.obr.plugin;
+
+import static aQute.bnd.build.Workspace.findWorkspace;
+import static aQute.lib.io.IO.getFile;
+import static biz.aQute.resolve.Bndrun.createBndrun;
+import static biz.aQute.resolve.obr.plugin.ObrExporter.TYPE;
+import static java.util.zip.GZIPInputStream.GZIP_MAGIC;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.shredzone.commons.xml.XQuery;
+
+import aQute.lib.io.IO;
+import biz.aQute.resolve.Bndrun;
+
+public final class ObrExporterTest {
+
+	private static Bndrun		project;
+	private static ObrExporter	underTest;
+
+	@BeforeClass
+	public static void setup() throws Exception {
+		project = createBndrun(findWorkspace(IO.work), getFile("test.bndrun"));
+        underTest = new ObrExporter();
+    }
+
+	@AfterClass
+	public static void teardown() throws IOException {
+		File testResource1 = IO.getFile("target/test.xml");
+		File testResource2 = IO.getFile("target/abc.xml");
+		File testResource3 = IO.getFile("target/abc.xml.gz");
+		File testResource4 = IO.getFile("obr");
+
+		IO.delete(testResource1.toPath());
+		IO.delete(testResource2.toPath());
+		IO.delete(testResource3.toPath());
+		IO.delete(testResource4.toPath());
+	}
+
+    @Test
+    public void test_obr_generation_without_name_configuration() throws Exception {
+        final Map<String, String> options = new HashMap<>();
+
+        underTest.export(TYPE, project, options);
+
+		assertThat(getFile("target/test.xml")).exists();
+    }
+
+    @Test
+    public void test_obr_generation_with_name_configuration() throws Exception {
+		final Map<String, String> options = createOptions("name", "abc.xml");
+
+        underTest.export(TYPE, project, options);
+
+		assertThat(getFile("target/abc.xml")).exists();
+    }
+
+    @Test
+    public void test_obr_generation_without_system_bundle_capabilities() throws Exception {
+        final Map<String, String> options = new HashMap<>();
+
+        underTest.export(TYPE, project, options);
+
+		final File obr = getFile("target/test.xml");
+        assertThat(obr).exists();
+
+        final XQuery xq = XQuery.parse(new FileReader(obr));
```

Comment:
Is there really any reason to use XPath here? The XML could be read back by bnd into a repository and queried using existing API. That would also better validate that the XML wasn't mangled.

Reply:
In fact if the XQuery usage here is removed then the whole dependency becomes unnecessary.

Reply:
I actually thought to make use of an external library for XML parsing as it would make the code more readable, otherwise, the code will be very cluttered with SAXParser.

Reply:
I agree that we should not use a dependency unless it is paramount. You can also parse it with XPath, which is built in.

Reply:
> I actually thought to make use of an external library for XML parsing as it would make the code more readable, otherwise, the code will be very cluttered with SAXParser.

The bnd code already includes a class for parsing an XML into a repository. There would be no need to create a parser for the test.

Reply:
> The bnd code already includes a class for parsing an XML into a repository. There would be no need to create a parser for the test.

Ah that would be awesome. I didn't know about it. Could you kindly point me to the API?

Reply:
I have updated the code to use `XPathFactory` instead which doesn't require use of any external library as well.

Reply:
> Ah that would be awesome. I didn't know about it. Could you kindly point me to the API?

[XMLResourceParser](https://github.com/bndtools/bnd/blob/8b7d203e5d87026651d7cbf88a75903ef59d0de6/biz.aQute.bndlib/src/aQute/bnd/osgi/repository/XMLResourceParser.java) usage visible in [RepositoryTest](https://github.com/bndtools/bnd/blob/8b7d203e5d87026651d7cbf88a75903ef59d0de6/biz.aQute.bndlib.tests/test/test/RepositoryTest.java)

This would be better than XPath as it demonstrates round tripping of the XML.

Reply:
@timothyjward Thanks a lot for the support! I have updated the tests to use `XMLResourceParser` instead. It looks pretty clean now.